packages/forms/src/__tests__/AppFormikProvider.spec.tsx

```diff
@@ -0,0 +1,95 @@
+import { MockEvent } from '@superdispatch/testutils';
+import { render, waitFor } from '@testing-library/react';
+import { FormikValues } from 'formik';
+import React from 'react';
+
+import { AppFormikContext, AppFormikProvider } from '../AppFormikProvider';
+import { AppFormikConfig, useAppFormik } from '../useAppFormik';
+
+async function renderProvider<TValues extends FormikValues, TResponse>(
+  defaultConfig: AppFormikContext,
+  formConfig: AppFormikConfig<TValues, TResponse>,
+) {
+  function Foo() {
+    const { handleSubmit, errors } = useAppFormik<TValues, TResponse>(
+      formConfig,
+    );
+    return (
+      <div>
+        {errors.name}
+        <button type="submit" onClick={() => handleSubmit()}>
+          Submit
+        </button>
+      </div>
+    );
+  }
+
+  return render(
+    <AppFormikProvider {...defaultConfig}>
+      <Foo />
+    </AppFormikProvider>,
+  );
+}
+
+test('default configs', async () => {
+  const getFormErrors = jest.fn((errorResponse) => errorResponse.fieldErrors);
+  const wrapper = await renderProvider(
+    { getFormErrors },
+    {
+      initialValues: { foo: '' },
+      onSubmit: () =>
+        Promise.reject({ fieldErrors: { name: 'Name is Required.' } }),
+    },
+  );
+
+  MockEvent.click(wrapper.getByText('Submit'));
+
+  await waitFor(() => {
```

Comment:
it's better to `await wrapper.findByText('Name is Required.')` and then check how much time `getFormErrors` was called